openedx/core/djangoapps/user_api/accounts/tests/test_filters.py

```diff
@@ -0,0 +1,126 @@
+"""
+Test that various filters are fired for views in the certificates app.
+"""
+from django.http import HttpResponse
+from django.test import override_settings
+from openedx_filters import PipelineStep
+from openedx_filters.learning.filters import AccountSettingsRenderStarted
+from rest_framework import status
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+
+
+class TestRedirectToPageStep(PipelineStep):
+    """
+    Utility class used when getting steps for pipeline.
+    """
+
+    def run_filter(self, context):  # pylint: disable=arguments-differ
+        """Pipeline step that redirects to dashboard before rendering the account settings page."""
+
+        raise AccountSettingsRenderStarted.RedirectToPage(
+            "You can't access this page, redirecting to dashboard.",
+            redirect_to="/dashboard",
+        )
+
+
+class TestRenderCustomResponse(PipelineStep):
+    """
+    Utility class used when getting steps for pipeline.
+    """
+
+    def run_filter(self, context):  # pylint: disable=arguments-differ
+        """Pipeline step that returns a custom response when rendering the account settings page."""
+        response = HttpResponse("Here's the text of the web page.")
+        raise AccountSettingsRenderStarted.RenderCustomResponse(
+            "You can't access this page.",
+            response=response,
+        )
+
+class TestAccountSettingsRenderPipelineStep(PipelineStep):
+    """
+    Utility class used when getting steps for pipeline.
+    """
+
+    def run_filter(self, context):  # pylint: disable=arguments-differ
+        """Pipeline step that returns a custom response when rendering the account settings page."""
+        context += {
+            'test': 'test',
+        }
+        return context
+
+class TestAccountSettingsFilters(SharedModuleStoreTestCase):
```

Comment:
I'd use these tests as a template for testing this kind of filters:
```
test_account_render_filter_executed
test_account_render_alternative
test_account_redirect
test_account_redirect_default
test_account_custom_response
test_account_render_without_filter_config
```
Those are precisely the same as [here](https://github.com/openedx/edx-platform/blob/8b1c90a19d395bafbc2d76cbc35e1056f97df763/lms/djangoapps/courseware/tests/test_filters.py#L16). I know this needs to be documented somewhere -we're working on it- but again, you could use them as a guide.